### DISCUSSION TITLE
Default keys for translating labels

### DISCUSSION THREAD

**Discussion by axelbdt** [filamentphp/filament#6360] [Help]
Hello,

Thanks for making translation a first-class citizen in Filament.

As I am building my json translation file, I wonder if there is a default key for each translation (eg `'filament.resources.my-resource.fields.name'` for  that would render the `translateLabel` less needed and foster good practice to organize the translation files.

From reading the code, I _think_ there isn't, but curious if I have missed something.

Cheers.

**Comment by danharrin** [comment]
I agree, that is very good translation practice. I am incredibly against `translateLabel()` existing in the first place because it is bad practice. Maybe you could try and macro a `->translateResourceLabel()`? But I don't think calling `->label()` with the explicit string is too bad, it probably improves the clarity of your code.

**Comment by axelbdt** [comment]
Thanks for the reply, Dan.

I agree that `->label()` is a good solution, it just leaves the user to figure out himself how to organize his JSON file. I saw you making good points in other discussion about 1:1 English string translation being bad practice, but I think it might just be beginner's first impulse. Also, naming the strings is hard.

Could `getLabel`, `getModelLabel` first look into translation files for a key according to some convention (e.g. `filament.resources.user.phone_number`) and only fallback to capitalized English if this is not found? Users would just have to change their json, and not think about a key or add `->label`. It could effectively nudge them away from `->translateLabel`.

What do you think?

**Reply by danharrin** [reply]
Is there a way in Laravel to check if a translation string is resolvable instead of just returning the string itself again?

**Reply by axelbdt** [reply]
Laravel's Lang facade provides a `has` method apparently.
